File: Backend/src/main/java/org/example/datn_sd69/modules/concentration/service/ConcentrationService.java
package org.example.datn_sd69.modules.concentration.service;

import org.example.datn_sd69.entity.Concentration;
import org.example.datn_sd69.modules.concentration.dto.request.ConcentrationRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ConcentrationService {
    List<Concentration> getAll();
    Concentration getById(Integer id);
    Concentration create(ConcentrationRequest request);
    Concentration update(Integer id, ConcentrationRequest request);
    void delete(Integer id);

    Page<Concentration> getAll(String keyword, Pageable pageable);

    Page<Concentration> getActiveConcentrations(int page, int size);
}

File: Frontend/src/modules/admin/feature/concentration/services/concentration.service.ts
import api from '@/common/api';
import type { ConcentrationRequest } from '../types/concentration.type';

const API_URL = '/admin/concentrations';

export const concentrationService = {
  getAll(config?: any) {
    return api.get(API_URL, config);
  },
  
  create(data: ConcentrationRequest) {
    return api.post(API_URL, data);
  },
  
  update(id: number, data: ConcentrationRequest) {
    return api.put(`${API_URL}/${id}`, data);
  },
  
  delete(id: number) {
    return api.delete(`${API_URL}/${id}`);
  }
};